lib/trento/infrastructure/commanded/event_handlers/software_updates_discovery_event_handler.ex

```diff
@@ -0,0 +1,28 @@
+defmodule Trento.Infrastructure.Commanded.EventHandlers.SoftwareUpdatesDiscoveryEventHandler do
+  @moduledoc """
+  Event handler for software updates discovery related events.
+  Here is where the actual integration with the external system happens and relevant changes in Host are triggered.
+  """
+
+  use Commanded.Event.Handler,
+    application: Trento.Commanded,
+    name: "software_updates_discovery_event_handler"
+
+  require Logger
+
+  alias Trento.Hosts.Events.SoftwareUpdatesDiscoveryRequested
+
+  alias Trento.SoftwareUpdates.Discovery
+
+  def handle(
+        %SoftwareUpdatesDiscoveryRequested{
+          host_id: host_id,
+          fully_qualified_domain_name: fully_qualified_domain_name
+        },
+        _
+      ) do
+    Discovery.discover_host_software_updates(host_id, fully_qualified_domain_name)
+
+    :ok
```

Comment:
i find this always `:ok` return not really safe or robust.
I would personally check what the discovery returns, and based on that return some error.

We can implement the `error` function later on, even if it does a simple `:skip`.
This would give the look that we are dealing with the error (even though initially is doing nothing).

Maybe in the future, we could have some retry strategy, or broadcast some message to the frontend, or anything like that.

I find it a elegant solution at least.

PD: The [error](https://hexdocs.pm/commanded/Commanded.Event.Handler.html#callbacks) function

Reply:
Anyway, as we are not going to do anything with the failure by now, I'm okey to leaving as it is.
Maybe just putting a comment like `The error is not being handled, as there is not any strategy for it`

Reply:
I agree, some more robust error handling might be beneficial.
I just wanted to get the flow and iterate on errors as per the needs